src/explore-education-statistics-admin/src/pages/admin-dashboard/AdminDashboardPage.tsx

```diff
@@ -24,34 +26,40 @@ interface Model {
 const AdminDashboardPage = () => {
   const { user } = useAuthContext();
   const [model, setModel] = useState<Model>();
-  useEffect(() => {
-    Promise.all([
+  const reloadDashboard = useCallback(async () => {
+    const [draftReleases, scheduledReleases] = await Promise.all([
       dashboardService.getDraftReleases(),
       dashboardService.getScheduledReleases(),
-    ]).then(([draftReleases, scheduledReleases]) => {
-      const contactResultsByRelease = scheduledReleases.map(release =>
-        dashboardService
-          .getPreReleaseContactsForRelease(release.id)
-          .then(contacts => ({
-            releaseId: release.id,
-            contacts,
-          })),
-      );
+    ]);
+
+    const contactResultsByRelease = scheduledReleases.map(release =>
+      dashboardService
+        .getPreReleaseContactsForRelease(release.id)
+        .then(contacts => ({
+          releaseId: release.id,
+          contacts,
+        })),
+    );
+
+    const contactResults = await Promise.all(contactResultsByRelease);
+
+    const preReleaseContactsByScheduledRelease: Dictionary<PrereleaseContactDetails[]> = {};
 
-      return Promise.all(contactResultsByRelease).then(contactResults => {
-        const preReleaseContactsByScheduledRelease: Dictionary<PrereleaseContactDetails[]> = {};
-        contactResults.forEach(result => {
-          const { releaseId, contacts } = result;
-          preReleaseContactsByScheduledRelease[releaseId] = contacts;
-        });
-        setModel({
-          draftReleases,
-          scheduledReleases,
-        });
-      });
+    contactResults.forEach(result => {
+      const { releaseId, contacts } = result;
+      preReleaseContactsByScheduledRelease[releaseId] = contacts;
```

Comment:
What are we actually doing with this? I can't see where it's being used, but it's mutative so it fires off a red flag in my mind anyway.

Reply:
This work predates this PR - agree it should be looked at sometime though!